### PR TITLE
fix(docs): Add required S3 PutObjectTagging permission to IAM policy in README

### PR DESCRIPTION
**Pull request by chrisRedwine**
Resolves https://github.com/vmware-tanzu/velero/issues/8062

Also, see this [issue](https://github.com/terraform-aws-modules/terraform-aws-iam/issues/518) and this [PR](https://github.com/terraform-aws-modules/terraform-aws-iam/pull/517) for more details.

### CODE REVIEW

File: backupstoragelocation.md [hunk 1]
```diff
@@ -112,7 +112,8 @@ spec:
     enableSharedConfig: "true"
 
     # Tags that need to be placed on AWS S3 objects. 
-    # For example "Key1=Value1&Key2=Value2"
+    # For example, "Key1=Value1&Key2=Value2".
+    # This requires the IAM policy to have the "s3:PutObjectTagging" permission on the S3 bucket.
```
Comment (shubham-pampattiwar) [inline backupstoragelocation.md:116]: you can remove the permission note from here.

**Review by sseago — APPROVED** [review]

**Comment by sseago** [comment]
@chrisRedwine DCO check failed -- commits need to be signed off
https://velero.io/docs/main/code-standards/#dco-sign-off

**Review by shubham-pampattiwar — APPROVED** [review]

**Comment by chrisRedwine** [comment]
Hi @sseago - just curious, is this waiting on anything further from my end? Thanks

**Comment by sseago** [comment]
@chrisRedwine No, it's all good here. It's got the necessary acks already -- I can merge it now.

**Pull request merged**